qiskit_experiments/curve_analysis/standard_analysis/error_amplification_analysis.py

```diff
@@ -128,15 +147,43 @@ def _generate_fit_guesses(
             user_opt.p0.set_if_empty(amp=max_y - min_y)
             user_opt.bounds.set_if_empty(amp=(-2 * max_abs_y, 2 * max_abs_y))
 
-        # Base the initial guess on the intended angle_per_gate.
-        angle_per_gate = self._get_option("angle_per_gate")
-
-        if angle_per_gate is None:
-            raise CalibrationError("The angle_per_gate was not specified in the analysis options.")
+        # Base the initial guess on the intended angle_per_gate and phase offset.
+        apg = self._get_option("angle_per_gate")
+        phi = self._get_option("phase_offset")
+
+        # Prepare logical guess for specific condition (often satisfied)
+
+        # The fit function can be transformed into
+        # y = A cos((d_theta + apg) x - phi)
+        #   = A cos(d_theta x)cos(apg x - phi) - A sin(d_theta x)sin(apg x - phi)
+        # If apg x - phi = (2n + 1) pi / 2 is satisfied,
+        # y = ∓ A sin(d_theta x) ~ ∓ A d_theta x, when d_theta x << 1
+        # Finally, d_theta = ∓ y / (A x)
+
+        d_theta_guesses = []
+
+        offsets = apg * curve_data.x + phi
+        amp = user_opt.p0.get("amp", self._get_option("amp"))
+        for i in range(curve_data.x.size):
+            xi = curve_data.x[i]
+            yi = curve_data.y[i]
+            if np.isclose(offsets[i] % np.pi, np.pi / 2) and xi > 0:
+                # Condition satisfied: i.e. cos(apg x + phi) = 0
+                err = -np.sign(np.sin(offsets[i])) * (yi - user_opt.p0["base"]) / (0.5 * amp)
+                # Validate estimate. This is first order term of Maclaurin expansion.
+                if np.abs(err) < 0.5:
+                    d_theta_guesses.append(err / xi)
+                else:
+                    # Terminate guess generation because larger d_theta x will start to
+                    # reduce net y value and underestimate the rotation.
+                    break
+
+        # Add naive guess for more coverage
+        guess_range = max(abs(apg), np.pi / 2)
+        d_theta_guesses.extend(np.linspace(-guess_range, guess_range, 21))
```

Comment:
How many guesses do we now have? I'm guessing at most `len(data.x) + 21` is this too much? Perhaps we can reduce the 21 to e.g. 11 to gain a bit of speed?

Reply:
21 is from your original default analysis options. I'll test if all tests pass with 11.